src/containers/Main/AuthenticatedWrapper.tsx

```diff
@@ -4,17 +4,26 @@ import { useRouter } from '../../utils/hooks/useRouter'
 import isLoggedIn from '../../utils/helpers/loginCheck'
 import NonRegisteredLogin from '../User/NonRegisteredLogin'
 import { Redirect } from 'react-router'
+import { Loading } from '../../components'
 
 const AuthenticatedContent: React.FC = () => {
   const {
-    location: { pathname, search },
+    location,
     query: { sessionId },
   } = useRouter()
   if (isLoggedIn()) return <SiteLayout />
   // If there is a sessionId in the URL, then need to login as nonRegistered
   // before continuing
-  if (sessionId) return <NonRegisteredLogin option="redirect" redirect={pathname + search} />
-  return <Redirect to={{ pathname: '/login', state: { from: location.pathname } }} />
+
+  // Location is not fully loaded unless 'state' is present, TODO try upgrading react router
+  // without this sessionId is missing when navigating via Click Here in forgot password verification
+  if ('state' in location) {
+    let { pathname, search } = location
+    if (sessionId) return <NonRegisteredLogin option="redirect" redirect={pathname + search} />
+    return <Redirect to={{ pathname: '/login', state: { from: location.pathname } }} />
+  }
```

Comment:
Cool, so this gets updated from "location" props as soon as "state" becomes available, and then the component re-renders?

Looks good, thanks. I can't really test this locally as it seems to work regardless, it's only on the server that it seems to be a problem. I'll make a new build and include this and see how it goes.

Reply:
Hmm, didn't seem to fix it.

Will look into it more later